internal/lru/lru.go

```diff
@@ -34,61 +38,96 @@ type Cache struct {
 	// executed when an entry is purged from the cache.
 	OnEvicted func(key string, value interface{})
 
-	ll    *list.List
-	cache map[string]*list.Element
+	lock   sync.RWMutex
+	ll     *list.List
+	cache  map[string]*list.Element
+	window time.Duration
 }
 
 type entry struct {
-	key   string
-	value interface{}
+	key       string
```

Comment:
Why a key is needed?

Reply:
We use the key to delete the entries from the map when they get evicted.